cms/djangoapps/contentstore/views/tests/test_videos.py

```diff
@@ -854,6 +857,171 @@ def test_video_image_validation_message(self, image_data, error_message):
+                'preferred_languages': ['es', 'ur']
+            },
+            'Invalid languages.'
+        ),
+        (
+            {
+                'provider': TranscriptProvider.THREE_PLAY_MEDIA
+            },
+            'Invalid 3play turnaround.'
+        ),
+        (
+            {
+                'provider': TranscriptProvider.THREE_PLAY_MEDIA,
+                'three_play_turnaround': 'default'
+            },
+            'Invalid languages.'
+        ),
+        (
+            {
+                'provider': TranscriptProvider.THREE_PLAY_MEDIA,
+                'three_play_turnaround': 'default',
+                'preferred_languages': ['es', 'ur']
+            },
+            'Invalid languages.'
+        ),
+        # Success
+        (
+            {
+                'provider': TranscriptProvider.CIELO24,
+                'cielo24_fidelity': 'PROFESSIONAL',
+                'cielo24_turnaround': 'STANDARD',
+                'preferred_languages': ['en']
+            },
+            ''
+        ),
+        (
+            {
+                'provider': TranscriptProvider.THREE_PLAY_MEDIA,
+                'three_play_turnaround': 'default',
+                'preferred_languages': ['en']
+            },
+            ''
+        )
+    )
+    @ddt.unpack
+    @override_switch(THIRD_PARTY_TRANSCRIPTION_ENABLED, True)
+    def test_video_transcript(self, preferences, error_message):
+        """
+        Tests that transcript handler worrks correctly.
+        """
+        video_transcript_url = self.get_url_for_course_key(self.course.id)
+        response = self.client.post(
+            video_transcript_url,
+            json.dumps({
+                'provider': preferences.get('provider', ''),
+                'cielo24_fidelity': preferences.get('cielo24_fidelity', ''),
+                'cielo24_turnaround': preferences.get('cielo24_turnaround', ''),
+                'three_play_turnaround': preferences.get('three_play_turnaround', ''),
+                'preferred_languages': preferences.get('preferred_languages', []),
+            }),
+            content_type='application/json'
+        )
+        response = json.loads(response.content)
+        if error_message:
+            self.assertEqual(response['status'], 400)
+            self.assertEqual(response['error'], error_message)
+        else:
+            self.assertTrue('transcript_preferences' in response)
+
+    @ddt.data(
+        None,
+        {
+            'provider': TranscriptProvider.CIELO24,
+            'cielo24_fidelity': 'PROFESSIONAL',
+            'cielo24_turnaround': 'STANDARD',
+            'preferred_languages': ['en']
+        }
+    )
+    @override_settings(AWS_ACCESS_KEY_ID='test_key_id', AWS_SECRET_ACCESS_KEY='test_secret')
+    @patch('boto.s3.key.Key')
+    @patch('boto.s3.connection.S3Connection')
+    def test_transcript_preferences_metadata(self, transcript_preferences, mock_conn, mock_key):
```

Comment:
I've seen this and existing testing doing a lot more other things which we don't need. So, leaving it as it.